### PR TITLE
Added more information for Shopify Product

### PR DESCRIPTION
**Pull request by jerairrest**
Thanks for bootstraping this project! I look forward to working on it. So far I've added more information for Product as well as Variants. As I need more, I'll be adding more Shopify Entities as well as Tests. Let me know if you have any questions or comments.

### CODE REVIEW

File: product.go [hunk 1]
```diff
@@ -23,8 +23,28 @@ type ProductServiceOp struct {
...
-	Title string `json:"title"`
+	ID             int             `json:"id"`
+	Title          string          `json:"title"`
+	BoldHTML       string          `json:"body_html"`
+	Vendor         string          `json:"vendor"`
+	ProductType    string          `json:"product_type"`
+	Handle         string          `json:"handle"`
+	CreatedAt      string          `json:"created_at"`
+	UpdatedAt      string          `json:"updated_at"`
+	PublishedAt    string          `json:"published_at"`
```
Comment (dlebech) [inline product.go]: Could you add these three dates as `*time.Time` instead, similar to the [order dates](https://github.com/getconversio/go-shopify/blob/64825aa463f6dd5c7e81a0e15a402f807178e117/order.go#L50-L53)?

File: product.go [hunk 2]
```diff
@@ -23,8 +23,28 @@ type ProductServiceOp struct {
 
 // Product represents a Shopify product
 type Product struct {
-	ID    int    `json:"id"`
-	Title string `json:"title"`
+	ID             int             `json:"id"`
+	Title          string          `json:"title"`
+	BoldHTML       string          `json:"body_html"`
```
Comment (dlebech) [inline product.go]: Any reason for "renaming" this field instead of `BodyHTML` ?

File: variant.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	Grams                int    `json:"grams"`
+	InventoryPolicy      string `json:"inventory_policy"`
+	CompareAtPrice       string `json:"compare_at_price"`
+	FulfillmentService   string `json:"fulfillment_service"`
+	InventoryManagement  string `json:"inventory_management"`
+	Option1              string `json:"option1"`
+	Option2              string `json:"option2"`
+	Option3              string `json:"option3"`
+	CreatedAt            string `json:"created_at"`
+	UpdatedAt            string `json:"updated_at"`
```
Comment (dlebech) [inline variant.go]: Same comment here for the dates as above.

**Review by dlebech — CHANGES_REQUESTED** [review]
Thank you so much for the PR @jerairrest! I have a few comments for the PR that I hope you agree with.

Looking forward to further PRs from you :-)

File: product.go (hunk 2 above)
Comment (jerairrest) [reply product.go]: Hahaha wow sorry, I was exhausted when I put this PR in.

File: variant.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// Variant represents a Shopify variant
+type Variant struct {
+	ID                   int        `json:"id"`
+	ProductID            int        `json:"product_id"`
+	Title                string     `json:"title"`
+	Sku                  string     `json:"sku"`
+	Position             int        `json:"position"`
+	Grams                int        `json:"grams"`
+	InventoryPolicy      string     `json:"inventory_policy"`
+	CompareAtPrice       string     `json:"compare_at_price"`
```
Comment (dlebech) [inline variant.go]: Sorry for the extra comments:

1. This is probably a decimal right? If so, you can use `*decimal.Decimal` like the order [`amount`](https://github.com/getconversio/go-shopify/blob/64825aa463f6dd5c7e81a0e15a402f807178e117/order.go#L100)
2. I noticed that you didn't include the [`price`](https://help.shopify.com/api/reference/product_variant#price-property) field anywhere. I think that would be a good addition to this resource.

**Review by dlebech — CHANGES_REQUESTED** [review]
Thanks for the first change. I noticed another small thing here that would be good to change.

File: variant.go (hunk 4 above)
Comment (jerairrest) [reply variant.go]: Good catch! added and changed the typing.

**Review by dlebech — APPROVED** [review]
Awesome thanks so much!

**Comment by dlebech** [comment]
@jerairrest Thanks for the contribution :beer:

**Pull request merged**